### PR TITLE
Add native macOS support

### PR DESCRIPTION
**Pull request by JadTala**
This adds a native macOS platform layer using Cocoa with C and Objective C as well as Makefiles and updated scripts.

It has been tested on two Macs without issues.

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -11,15 +11,14 @@ Forums: https://forums.kohiengine.com
 Kohi will be a 3D engine from the start, with most of it built from scratch with explanations along the way. The series starts off as more of a tutorial to get up and running, with explanations becoming higher level as things progress to keep the pacing up.
 
 ## .platform support
-Windows and Linux are both supported from the start, with native (non-GLFW) Mac support to be added in the future. Android and iOS runtime support may also be added down the road.
+Windows, Linux and macOS are supported. Android and iOS runtime support may also be added down the road.
```
Comment (travisvroman) [inline readme.md:14]: Thank you for the hard work and for submitting this! It took me quite a while to get the chance to review all of it, but here we are.

Please note that since I do not currently own a Mac, I have no way to verify the workings of this and I'll take your word that it works properly for now. The code all looks fine to me, so I'll merge it with one minor change here to the readme:

Please reword this to "Windows and Linux are both officially supported, with unofficial macOS support added by the community." (or, feel free to add your username in there if you wish since you deserve the credit) for now. I will perform the merge once this change has been made.

Once I have an environment, I can _actually_ execute this in, I'll likely do another pass over this code and then stamp support as official at that time.

Thank you!

**Review by travisvroman — CHANGES_REQUESTED** [review]
Please see requested changes in the readme before this can be merged. Thank you for submitting this!

File: readme.md (hunk 1 above)
Comment (JadTala) [reply readme.md]: Thank you for reviewing this.

I've updated the readme with your rewording as I consider contributing to be enough credit. 

I'd sold my Mac when I updated my fork to the new memory model so I couldn't test it. However I just got an M1 MacBook Air so I fixed compilation issues and added a specific macOS post build script in the two additional commits.

Therefore this also works natively on Apple Silicon.

**Comment by JadTala** [comment]
Done, thanks for reviewing it!

File: readme.md (hunk 1 above)
Comment (travisvroman) [reply readme.md]: Awesome, looks good. Thanks again for this! Merging.
